zilliqa/src/api/eth.rs

```diff
@@ -423,14 +423,21 @@ fn get_logs(params: Params, node: &Arc<Mutex<Node>>) -> Result<Vec<eth::Log>> {
             .get_block(block_hash)?
             .ok_or_else(|| anyhow!("block not found"))?))),
         (None, from, to) => {
-            let from = node
+            let Some(from) = node
                 .resolve_block_number(from.unwrap_or(BlockNumberOrTag::Latest))?
-                .unwrap()
-                .number();
-            let to = node
+                .as_ref()
+                .map(Block::number)
+            else {
+                return Ok(vec![]);
+            };
+
+            let Some(to) = node
                 .resolve_block_number(to.unwrap_or(BlockNumberOrTag::Latest))?
-                .unwrap()
-                .number();
+                .as_ref()
+                .map(Block::number)
+            else {
+                return Ok(vec![]);
+            };
```

Comment:
I'm not actually sure what the intended behaviour is (maybe we should check on an Ethereum testnet), but this will have the affect that if `to` is too large, I will always get an empty response which is not what I would immediately assume. I would hope instead that the block range is instead truncated to a valid range.

i.e. if the network was at block height `N` and I made a request for logs between `N - 10` and `N + 10`, I would assume I would receive the logs between `N - 10` and `N` rather than nothing.

Best to check another implementation or API I think?

Reply:
Having just tested this I can confirm that what you suggested is correct, and have pushed changes to implement this behaviour.